File: Valid Parentheses - Leetcode 20/Valid Parentheses - Leetcode 20.js
var isValid = function(s) {
    const hashmap = { ')': '(', '}': '{', ']': '[' };
    const stk = [];

    for (const c of s) {
        if (!(c in hashmap)) {
            stk.push(c);
        } else {
            if (stk.length === 0 || stk.pop() !== hashmap[c]) {
                return false;
            }
        }
    }

    return stk.length === 0;
};

File: Valid Parentheses - Leetcode 20/Valid Parentheses - Leetcode 20.cpp
#include <stack>
#include <unordered_map>
#include <string>
using namespace std;

class Solution {
public:
    bool isValid(string s) {
        unordered_map<char, char> hashmap = {{')', '('}, {'}', '{'}, {']', '['}};
        stack<char> stk;

        for (char c : s) {
            if (hashmap.find(c) == hashmap.end()) {
                stk.push(c);
            } else {
                if (stk.empty() || stk.top() != hashmap[c]) {
                    return false;
                }
                stk.pop();
            }
        }

        return stk.empty();
    }
};
